internal/run/run_stage_test.go

```diff
@@ -571,6 +596,50 @@ func (s *RunTestStage) all_exported_metrics_contain_label(labelName string, labe
 	return s
 }
 
+func (s *RunTestStage) the_trace_output_should_be_present() *RunTestStage {
+	s.require.NoError(s.tracerWriter.Close())
+	s.tracerWg.Wait()
+
+	s.require.NotEmpty(s.tracerOutput)
+	return s
+}
+
+func (s *RunTestStage) tracing_is_enabled() *RunTestStage {
+	s.tracerWg.Add(1)
+
+	go func() {
+		defer s.tracerWg.Done()
```

Comment:
What stops this executing immediately and copying nothing?

Reply:
`io.Copy` will block until EOF or error. 

**It is** actually executing immediately, but then just stops there and waits for the `Close()`of the `tracewriter` that will send the EOF

Reply:
Makes sense!